app/Http/Controllers/Service/Admin/VouchersController.php

```diff
@@ -47,20 +50,63 @@ public function create()
      */
     public function storeBatch(Request $request)
     {
-        $new_vouchers = [];
+        $newVouchers = [];
+
+        $sponsorIds = \App\Sponsor::all()->pluck('id')->toArray();
+
+        $voucherRules = [
+            'sponsor_id' => [
+                'required',
+                Rule::in($sponsorIds)
```

Comment:
And you should store them it the Voucher class so they can be reused.
I think I already started there?

The start and end are part of this particular form, but not the sponsor_id.